src/convex-fed/ConvexFedV2.sol

```diff
@@ -0,0 +1,260 @@
+
+        //Withdraw and unwrap curveLP tokens from convex, but don't claim rewards
+        require(baseRewardPool.withdrawAndUnwrap(crvLpNeeded, false), "Convex withdraw failed");
+
+        //Withdraw DOLA from curve pool
+        uint dolaWithdrawn = metapoolWithdraw(amountDola, maxLossWithdrawBps);
+        require(dolaWithdrawn > 0, "Must contract");
+        uint burnAmount = _burnAndPay();
+        emit Contraction(burnAmount);
+    }
+
+    /**
+    * @notice Withdraws every remaining crvLP token. Can take up to maxLossWithdrawBps in loss, compared to dolaSupply.
+    * It will still be necessary to call takeProfit to withdraw any potential rewards.
+    */
+    function contractAll() external onlyRole(chair){
+        baseRewardPool.withdrawAllAndUnwrap(false);
+        uint dolaMinOut = dolaSupply * (10_000 - maxLossWithdrawBps) / 10_000;
+        crvMetapool.remove_liquidity_one_coin(crvLpSupply(), 0, dolaMinOut);
+        uint burnAmount = _burnAndPay();
+        emit Contraction(burnAmount);
+    }
+
+
+    /**
+    * @notice Withdraws the profit generated by convex staking to governance
+    */
+    function claimRewards() public {
+        require(baseRewardPool.getReward());
+        crv.transfer(gov, crv.balanceOf(address(this)));
+        cvx.transfer(gov, cvx.balanceOf(address(this)));
+    }
+
+    function claimOther(address otherReward) external {
+        require(otherReward != address(dola), "Cant claim dola");
+        require(otherReward != address(crvMetapool), "Cant claim crv LP tokens");
+        IERC20(otherReward).transfer(gov, IERC20(otherReward).balanceOf(address(this)));
+    }
+
+    /**
+    * @notice Burns amount of dola supply with a maximum of the entire dola supply.
+    * @param amount Amount of dola supply to be burnt.
+    */
+    function burnDolaSupply(uint amount) external {
+        if(amount > dolaSupply){
+            dola.transferFrom(msg.sender, address(this), dolaSupply);
+        } else {
+            dola.transferFrom(msg.sender, address(this), amount);
+        }
+        _burnAndPay();
+    }
+
+    /**
+    * @notice Burns all dola tokens held by the fed up to the dolaSupply, taking any surplus as profit.
+    */
+    function _burnAndPay() internal returns(uint burnAmount){
+        uint dolaBal = dola.balanceOf(address(this));
+        if(dolaBal > dolaSupply){
+            IERC20(dola).transfer(gov, dolaBal - dolaSupply);
+            burnAmount = dolaSupply;
+            dolaSupply = 0;
+        } else {
+            burnAmount = dolaBal;
+            dolaSupply -= dolaBal;
+        }
+        IERC20(dola).burn(burnAmount);
+    }
+
+    /**
+    * @notice Withdraws all owned curve LP tokens to governance.
+    * @dev Only to be used in emergency situations and with full consent of governance.
+    * @param claim Claim convex staking rewards if true, otherwise withdraw without claiming.
+    */
+    function emergencyWithdraw(bool claim) external onlyRole(gov) {
```

Comment:
This function mostly exists in case something breaks with the normal contraction behaviour. We're also not too keen on having msigs be able to withdraw crv LP tokens, even if they'd ultimately be withdrawn to being under governance control.